src/stores/provider.ts

```diff
@@ -0,0 +1,150 @@
+import { capital } from 'case'
+import { defineStore } from 'pinia'
+import { computed, reactive, Ref, ssrRef } from '@nuxtjs/composition-api'
+import axios from 'axios'
+
+import {
+  ALL_MEDIA,
+  AUDIO,
+  IMAGE,
+  SupportedMediaType,
+  supportedMediaTypes,
+} from '~/constants/media'
+import { warn } from '~/utils/console'
+import { useSearchStore } from '~/stores/search'
+
+import { providerServices } from '~/data/media-provider-service'
+import type { MediaProvider } from '~/models/media-provider'
+import { FetchState, useFetchState } from '~/composables/use-fetch-state'
+
+export interface ProviderState {
+  providers: {
+    audio: MediaProvider[]
+    image: MediaProvider[]
+  }
+  fetchState: {
+    audio: FetchState
+    image: FetchState
+  }
+}
+/**
+ * Sorts providers by their source_name property.
+ * @param data - initial unordered list of providers
+ */
+const sortProviders = (data: MediaProvider[]): MediaProvider[] => {
+  return [...data].sort((sourceObjectA, sourceObjectB) => {
+    const nameA = sourceObjectA.source_name.toUpperCase()
+    const nameB = sourceObjectB.source_name.toUpperCase()
+    return nameA.localeCompare(nameB)
+  })
+}
+/**
+ * Timestamp is used to limit the update frequency to one every 60 minutes per request.
+ */
+const lastUpdated: Ref<Date | null> = ssrRef(null)
+
+export const useProviderStore = defineStore('provider', () => {
+  const fetchStates = {
+    [AUDIO]: useFetchState(),
+    [IMAGE]: useFetchState(),
+  }
+
+  const state: ProviderState = reactive({
+    providers: {
+      [AUDIO]: [],
+      [IMAGE]: [],
+    },
+    fetchState: {
+      audio: fetchStates[AUDIO].fetchState,
+      image: fetchStates[IMAGE].fetchState,
+    },
+  })
+
+  /**
+   * Returns the display name for provider if available, or capitalizes the given providerCode.
+   *
+   * @param providerCode - the `source_name` property of the provider
+   */
+  const getProviderName = (providerCode: string) => {
+    const searchStore = useSearchStore()
+    const mediaType =
+      searchStore.searchType === ALL_MEDIA ? IMAGE : searchStore.searchType
+    const matchingProviders = state.providers[mediaType].filter(
+      (p) => p.source_name === providerCode
+    )
+    return matchingProviders.length
+      ? matchingProviders[0].display_name || capital(providerCode)
+      : capital(providerCode)
+  }
+
+  /**
+   * Fetch providers only if there is no data, or if the last update for current request
+   * was more than 1 hour ago.
+   */
+  const needsUpdate = computed(() => {
+    const noData = supportedMediaTypes.some(
+      (mediaType) => !state.providers[mediaType].length
+    )
+    if (noData || !lastUpdated.value) {
+      return true
+    }
+
+    const timeSinceLastUpdate =
+      new Date().getTime() - new Date(lastUpdated.value).getTime()
+    return timeSinceLastUpdate > 3600000
```

Comment:
I believe that number could be an environment variable. Just a thought, is not really a requirement right now.

Reply:
Updated in [150b8f8](https://github.com/WordPress/openverse-frontend/pull/1158/commits/150b8f81239e492a01136a80ab603aa001515316)